### PR TITLE
137 create a function for container name creation logic

### PR DESCRIPTION
**Pull request by Francois-Werbrouck**
ref #137

### CODE REVIEW

**Comment by Francois-Werbrouck** [comment]
I haven't write 
>unit tests for the new function to ensure it behaves correctly.

I will write unittest to make sure the 2 functions behave as expected. Previous tests should tell us if the changes are working for the refactoring.

**Review by k-allagbe — APPROVED** [review]
There might still be manual building of those strings in:
- `tests/test_azure_storage.py`
- `tests/fertiscan/test_datastore.py`

**Comment by Francois-Werbrouck** [comment]
I will rebase on main and check if any tests are having issues atm. I will then add new tests where it is necessary and update the old tests with the new functions!

File: datastore/blob/azure_storage_api/__init__.py [hunk 1]
```diff
@@ -50,6 +50,37 @@ async def generate_hash(image):
         print(error.__str__())
         raise Exception("Unhandeled Datastore.blob.azure_storage Error")
 
+def build_container_name(name:str,tier:str="user"):
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:53]: The spacing on the types does not look right. Are we running ruff on these files?

**Comment by Francois-Werbrouck** [comment]
workflow tests are not passing due to our new DB server being needy

File: datastore/__init__.py [hunk 2]
```diff
@@ -81,10 +81,14 @@ async def new_user(cursor, email, connection_string, tier="user") -> User:
         blob_service_client = BlobServiceClient.from_connection_string(
             connection_string
         )
-        container_client = blob_service_client.create_container(f"{tier}-{user_uuid}")
+        container_client = blob_service_client.create_container(
+            azure_storage.build_container_name(str(user_uuid), "user")
```
Comment (ChromaticPanic) [inline datastore/__init__.py:85]:
```
Defined in datastore/db/queries/user/__init__.py
def register_user(cursor, email: str) -> None:
```

```Line 79 user_uuid = user.register_user(cursor, email)```

Currently register_user is defined to return None, can we update that to str?

File: datastore/blob/azure_storage_api/__init__.py [hunk 3]
```diff
@@ -50,6 +69,40 @@ async def generate_hash(image):
         print(error.__str__())
         raise Exception("Unhandeled Datastore.blob.azure_storage Error")
 
+
+def build_container_name(name: str, tier: str = "user"):
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:73]: Needs return type

File: datastore/blob/azure_storage_api/__init__.py [hunk 4]
```diff
@@ -50,6 +69,40 @@ async def generate_hash(image):
...
+    Parameters:
+    - name (str): the name of the container. Usually the user uuid
+    - tier (str): the tier of the container; Default is 'user'
+    """
+    if not name:
+        raise ValueError("Name is required")
+    return "{}-{}".format(tier, name)
+
+
+def build_blob_name(folder_path: str, blob_name: str, file_type: str = None):
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:87]: Needs return type

File: datastore/blob/azure_storage_api/__init__.py [hunk 5]
```diff
@@ -50,6 +69,40 @@ async def generate_hash(image):
...
+    We include a tier to better structure the container names in the future. Other tiers could be 'dev' or 'test-user'
+
+    Parameters:
+    - name (str): the name of the container. Usually the user uuid
+    - tier (str): the tier of the container; Default is 'user'
+    """
+    if not name:
+        raise ValueError("Name is required")
+    return "{}-{}".format(tier, name)
+
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:85]: Might want to check for ""

File: datastore/blob/azure_storage_api/__init__.py [hunk 6]
```diff
@@ -50,6 +69,40 @@ async def generate_hash(image):
...
+    """
+    if not folder_path:
+        raise ValueError("Folder name is required")
+    if not blob_name:
+        raise ValueError("Image uuid is required")
+    if file_type is not None:
+        return "{}/{}.{}".format(folder_path, blob_name, file_type)
+    else:
+        return "{}/{}".format(folder_path, blob_name)
+
```
Comment (ChromaticPanic) [inline datastore/blob/azure_storage_api/__init__.py:104]: might want to check for "" here too

**Review by ChromaticPanic — CHANGES_REQUESTED** [review]

File: datastore/__init__.py (hunk 2 above)
Comment (Francois-Werbrouck) [reply datastore/__init__.py:85]: made it a UUID instead

Comment (Francois-Werbrouck) [reply datastore/__init__.py:85]: This was mentionned as an issue in #201

**Review by ChromaticPanic — APPROVED** [review]

**Review by k-allagbe — APPROVED** [review]

**Pull request merged**